File: GSI QA testing tool/UI/UI_035_WorkHistoryVerification.cs
﻿using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Claims;
using System.Text;
using System.Threading.Tasks;

namespace GSI_QA_testing_tool.UI
{
    internal class UI_035_WorkHistoryVerification
    {
        public static void GoTo(IWebDriver driver)
        {
            Finder.ClickIt(driver, "//input[@id='ctl00_Main_content_Wizard1_StepNavigationTemplateContainerID_StepNextButton']");


            if (Data._Site.Contains("PUA"))
            {
                Finder.ClickIt(driver, "//label[@for='ctl00_Main_content_Wizard1_rblWorkHistoryVerify_1']");
            }
            else
            {
                if (Data._BasePeriod)
                {
                    Finder.ClickIt(driver, "//label[@for='ctl00_Main_content_Wizard1_rblWorkHistoryVerify_0']");
                }
                else
                {
                    Finder.ClickIt(driver, "//label[@for='ctl00_Main_content_Wizard1_rblWorkHistoryVerify_1']");
                }
            }
            Finder.ClickIt(driver, "//input[@id='ctl00_Main_content_Wizard1_StepNavigationTemplateContainerID_StepNextButton']");

        }

    }
    
}

File: GSI QA testing tool/UI/UI_035_WorkHistoryVerification.vb
Imports OpenQA.Selenium
Imports System
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Linq
Imports System.Security.Claims
Imports System.Text
Imports System.Threading.Tasks

Namespace GSI_QA_testing_tool.UI
    Friend Class UI_035_WorkHistoryVerification
        Public Shared Sub [GoTo](ByVal driver As IWebDriver, ByVal js As IJavaScriptExecutor)
            If Data._StopAt.Contains("Work History Verification") Then
                Data.StopAtDialog()
            End If

            If Finder.FindItByCSS(driver, "[for='ctl00_Main_content_Wizard1_rblWorkHistoryVerify_0']") Then
                Debug.WriteLine("WorkHistoryVerification is On")
            Else
                Debug.WriteLine("WorkHistoryVerification is Off")
                Return
            End If

            If Data._Site.Contains("PUA") Then
                Finder.JSClickIt(driver, "#ctl00_Main_content_Wizard1_rblWorkHistoryVerify_1", js)
            Else

                If Data._BasePeriod Then
                    Finder.JSClickIt(driver, "#ctl00_Main_content_Wizard1_rblWorkHistoryVerify_0", js)
                Else
                    Finder.JSClickIt(driver, "#ctl00_Main_content_Wizard1_rblWorkHistoryVerify_1", js)
                End If
            End If

            Finder.JSClickIt(driver, "#ctl00_Main_content_Wizard1_StepNavigationTemplateContainerID_StepNextButton", js)
        End Sub
    End Class
End Namespace
